tools/rvn/CommandLineApp.cs

```diff
@@ -51,6 +56,114 @@ public static int Run(string[] args)
             }
         }
 
+        private static void ConfigureSetupPackage()
+        {
+            _app.Command("create-setup-package", cmd =>
+            {
+                cmd.ExtendedHelpText = cmd.Description = "Creates RavenDB setup given setup-params.json";
+                cmd.HelpOption(HelpOptionString);
+                var setupParameters = ConfigureSetupParameters(cmd);
+                var packageOutputFileParameter = ConfigurePackageOutputFile(cmd);
+
+                cmd.OnExecuteAsync(async token =>
+                {
+                    var setupParam = setupParameters.Value();
+                    var packageOutParam = packageOutputFileParameter.Value();
+                    
+                    if (File.Exists(setupParam) == false)
+                        return ExitWithError("Path to setup params has not found", cmd);
+                    
+                    using (StreamReader file = File.OpenText(setupParam))
+                    {
+                        JsonSerializer serializer = new();
+                        var setupInfo = (SetupInfo)serializer.Deserialize(file, typeof(SetupInfo));
+                        if (packageOutParam == null) packageOutParam = setupInfo?.Domain + "-package.zip";
+                        else packageOutParam += "-package.zip";
+                        var settingsPath = setupParameters.Values[0];
+                        var setupLetsEncryptTask = await SetupLetsEncryptByRvn(setupInfo, settingsPath, new SetupProgressAndResult(), token);
```

Comment:
Does `create-setup-package` option allows to create other setup options? Not necessary now but in the future? `LetsEncrypt` is just one option, we also have option to provide own cert or run as unsecured.

IMO `create-setup-package` should accept the parameter which determines the mode of the setup operation e.g. `create-setup-package lets-encrypt .... ` 

I think this is something we need to thing about now because in the future we might want to expand it.

Reply:
Yeah I would expect that you can pass the select `Mode` in setup-parameters - e.g. if you come with your own certificate.